tilings/tiling.py

```diff
@@ -345,6 +346,128 @@ def _clean_assumptions(self) -> None:
                 res.append(ass)
         self._assumptions = tuple(sorted(set(res)))
 
+    @classmethod
+    def guess_from_gridded_perms(
+        cls, gps: Iterable[GriddedPerm], max_len: int = -1
+    ) -> "Tiling":
+        """Given a collection of gridded permutations, attempt to find the Tilings T
+        such that the collection of gridded permutations belong to Grid(T). The method
+        only looks for obstructions.
+
+        Raises:
+        ValueError: If the collection does not contain a pattern that is not avoided
+        by all.
+        """
+        gps = set(gps)
+        c, r, longest = cls._get_dimensions(gps)
+        max_len = longest if max_len == -1 else min(max_len, longest)
+        obs = cls._search(deque([GriddedPerm()]), gps, c, r, max_len)
+        return cls(obstructions=obs, requirements=(), assumptions=())
+
+    @staticmethod
+    def _extend_gp(gp: GriddedPerm, c: int, r: int) -> Iterator[GriddedPerm]:
```

Comment:
This should probably be turned into a method of `GriddedPerm`

Reply:
A similar function `GriddedPerm.insert_specific_point` might simplify this

Reply:
Fixed